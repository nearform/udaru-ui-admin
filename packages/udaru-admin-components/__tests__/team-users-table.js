import React from 'react'
import renderer from 'react-test-renderer'
import TeamUsersTable from 'team-users-table'

jest.mock('react-bootstrap-table', () => {
  return {
    BootstrapTable: props => {
      return (
        <table>
          {' // mocked table output'}
          {props.children}
        </table>
      )
    },
    TableHeaderColumn: props => {
      return (
        <tr>
          <th>{props.dataField + ' // mocked table header column'}</th>
        </tr>
      )
    }
  }
})

it('should render with props', () => {
  const props = {
    data: [],
    dataTotalSize: 100,
    sizePerPage: 5,
    onPageChange: () => {},
    sizePerPageList: [5, 10, 25, 50, 100],
    currentPage: 1,
    onSizePerPageList: () => {}
  }

  const component = renderer.create(<TeamUsersTable {...props} />)
  const tree = component.toJSON()

  expect(tree).toMatchSnapshot()
})
