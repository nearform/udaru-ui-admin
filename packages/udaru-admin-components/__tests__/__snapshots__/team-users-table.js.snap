// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render with props 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col-xs-12"
    >
      <h3>
        Users
      </h3>
    </div>
  </div>
  <div
    className="row"
  >
    <div
      className="col-xs-12"
    >
      <table>
         // mocked table output
        <tr>
          <th>
            id // mocked table header column
          </th>
        </tr>
        <tr>
          <th>
            name // mocked table header column
          </th>
        </tr>
      </table>
    </div>
  </div>
</div>
`;
